backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/repository/OrgChangeRepository.java

```diff
@@ -0,0 +1,614 @@
+	{
+		final OrgMappingId orgMappingId = orgMappingRepo.getCreateOrgMappingId(bpartnerId);
+		final BPartnerComposite bPartnerComposite = bPartnerCompositeRepo.getById(bpartnerId);
+
+		return OrgChangeBPartnerComposite.builder()
+				.bPartnerComposite(bPartnerComposite)
+				.bPartnerOrgMappingId(orgMappingId)
+				.membershipSubscriptions(getMembershipSubscriptions(bpartnerId, orgChangeDate, bPartnerComposite.getOrgId()))
+				.nonMembershipSubscriptions(getNonMembershipSubscriptions(bpartnerId, orgChangeDate, bPartnerComposite.getOrgId()))
+				.build();
+	}
+
+	private Collection<? extends FlatrateTerm> getNonMembershipSubscriptions(final BPartnerId bpartnerId, final LocalDate orgChangeDate, final OrgId orgId)
+	{
+		final Set<FlatrateTermId> flatrateTermIds = retrieveNonMembershipSubscriptions(bpartnerId, orgChangeDate, orgId);
+		return flatrateTermIds.stream()
+				.map(this::createFlatrateTerm)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	private List<FlatrateTerm> getMembershipSubscriptions(final BPartnerId bpartnerId, final LocalDate orgChangeDate, final OrgId orgId)
+	{
+		final Set<FlatrateTermId> membershipFlatrateTermIds = retrieveMembershipSubscriptions(bpartnerId, orgChangeDate, orgId);
+		return membershipFlatrateTermIds.stream()
+				.map(this::createFlatrateTerm)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	private FlatrateTerm createFlatrateTerm(@NonNull final FlatrateTermId flatrateTermId)
+	{
+		final I_C_Flatrate_Term term = flatrateDAO.getById(flatrateTermId);
+
+		final OrgId orgId = OrgId.ofRepoId(term.getAD_Org_ID());
+
+		final ZoneId timeZone = orgDAO.getTimeZone(orgId);
+
+		return FlatrateTerm.builder()
+				.flatrateTermId(flatrateTermId)
+				.orgId(orgId)
+				.billPartnerID(BPartnerId.ofRepoId(term.getBill_BPartner_ID()))
+				.billLocationId(BPartnerLocationId.ofRepoIdOrNull(term.getBill_BPartner_ID(), term.getBill_Location_ID()))
+				.shiToBPartnerId(BPartnerId.ofRepoIdOrNull(term.getDropShip_BPartner_ID()))
+				.shipToLocationId(BPartnerLocationId.ofRepoIdOrNull(term.getDropShip_BPartner_ID(), term.getDropShip_Location_ID()))
+				.productId(ProductId.ofRepoId(term.getM_Product_ID()))
+				.flartareConditionsId(ConditionsId.ofRepoId(term.getC_Flatrate_Conditions_ID()))
+				.isSimulation(term.isSimulation())
+				.status(FlatrateTermStatus.ofNullableCode(term.getContractStatus()))
+				.userId(UserId.ofRepoIdOrNull(term.getAD_User_InCharge_ID()))
+				.startDate(TimeUtil.asLocalDate(term.getStartDate(), timeZone))
+				.endDate(TimeUtil.asLocalDate(term.getEndDate(), timeZone))
+				.masterStartDate(TimeUtil.asLocalDate(term.getMasterStartDate(), timeZone))
+				.masterEndDate(TimeUtil.asLocalDate(term.getMasterEndDate(), timeZone))
+				.plannedQtyPerUnit(term.getPlannedQtyPerUnit())
+				.uomId(UomId.ofRepoIdOrNull(term.getC_UOM_ID()))
+				.build();
+	}
+
+	public void unmarkBillToDefaultLocations(final List<BPartnerLocation> locations)
```

Comment:
not sure why i didn'T write this in the first review, but: imho these methods don't really belong into a repo. i think i would move them to OrgChangeService

Reply:
Right. I missed them when moving.
Moved now.